### PR TITLE
Add development_platform for Unity crash reports

### PR DESCRIPTION
**Pull request by mrichards**
Crashlytics reports sent via DataTransport erroneously did not set the
development_platform field for Unity apps. This results in incorrect
internal metrics for Unity apps. b/171214452

### CODE REVIEW

**Comment by mrichards** [comment]
@mrwillis21 I'm looking for feedback on the overall approach, and if you think it is worth adding in the install_source field in a similar way.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCapture.java [hunk 1]
```diff
@@ -137,12 +137,20 @@ public Event captureEventData(
   }
 
   private CrashlyticsReport.Session.Application populateSessionApplicationData() {
-    return CrashlyticsReport.Session.Application.builder()
-        .setIdentifier(idManager.getAppIdentifier())
-        .setVersion(appData.versionCode)
-        .setDisplayVersion(appData.versionName)
-        .setInstallationUuid(idManager.getCrashlyticsInstallId())
-        .build();
+    CrashlyticsReport.Session.Application.Builder builder =
```
Comment (mrwillis21) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCapture.java:140]: For consistency, can we make this `final`?

**Review by mrwillis21 — APPROVED** [review]
One minor nit, but otherwise this seems great to me.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCapture.java (hunk 1 above)
Comment (mrichards) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCapture.java]: Done

**Pull request merged**